docs-source/spring/content/starters/ucp.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Oracle Spring Boot Starter for Universal Connection Pool"
+---
+
+This starter provides a connection (data source) to an Oracle Database using Universal Connection Pool, which provides an efficient way
+to use database connections. 
+
+To add this starter to your project, add this Maven dependency:
+
+```xml
+<dependency>
+    <groupId>com.oracle.database.spring</groupId>
+    <artifactId>oracle-spring-boot-starter-ucp</artifactId>
+    <version>23.4.0</version>
+    <type>pom</type>
+</dependency>
+```
+
+For Gradle projects, add this dependency:
+
+```
+implementation 'com.oracle.database.spring:oracle-spring-boot-starter-ucp:23.4.0:pom'
+```
+
+An Oracle data source will be injected into your application and can be used normally. You must configure
+it as shown below, and you should also add either Spring Data JDBC or Spring Data JPA to your project.
+
+To configure the datasource, provide a `spring.datasource` object in your Spring `application.yaml`,
+or equivalent, as shown in the example below.  The `oracleucp` entry is optional, and can be used
+to fine tune the configuration of the connection pool if desired.  For details of available settings,
+please refer to the [JavaDoc](https://docs.oracle.com/en/database/oracle/oracle-database/21/jjuar/oracle/ucp/jdbc/UCPDataSource.html).
+
+```yaml
+spring:
+  jpa:
+    hibernate:
+      ddl-auto: validate
+    properties:
+      hibernate:
+        dialect: org.hibernate.dialect.OracleDialect
+        format_sql: true
+    show-sql: true
+  datasource:
+    url: jdbc:oracle:thin:@//myhost:1521/pdb1
+    username: account
+    password: account
+    driver-class-name: oracle.jdbc.OracleDriver
+    type: oracle.ucp.jdbc.PoolDataSource
+    oracleucp:
+      connection-factory-class-name: oracle.jdbc.pool.OracleDataSource
+      connection-pool-name: AccountConnectionPool
+      initial-pool-size: 15
+      min-pool-size: 10
+      max-pool-size: 30
+```
+
+The `spring.datasource.url` can be in the basic format (as shown above), or in TNS format if your application
```

Comment:
"... format (as previously shown), or ..."

Use:
Transparent Network Substrate (TNS)

after which you can use TNS